torch/_export/serde/serialize.py

```diff
@@ -1720,3 +1729,313 @@ def deserialize(
             )
         )
     )
+
+
+def _canonicalize_graph(sorted_inputs, sorted_outputs, graph) -> Graph:
+    def _get_argument(a: Argument):
+        if a.as_none is not None:
+            return None
+        elif a.as_tensor is not None:
+            return a.as_tensor
+        elif a.as_tensors is not None:
+            return a.as_tensors
+        elif a.as_int is not None:
+            return None
+        elif a.as_ints is not None:
+            return None
+        elif a.as_float is not None:
+            return None
+        elif a.as_floats is not None:
+            return None
+        elif a.as_string is not None:
+            return None
+        elif a.as_strings is not None:
+            return None
+        elif a.as_sym_int is not None:
+            return a.as_sym_int
+        elif a.as_sym_ints is not None:
+            return a.as_sym_ints
+        elif a.as_scalar_type is not None:
+            return None
+        elif a.as_memory_format is not None:
+            return None
+        elif a.as_device is not None:
+            return None
+        elif a.as_bool is not None:
+            return None
+        elif a.as_bools is not None:
+            return None
+        elif a.as_sym_bool is not None:
+            return a.as_sym_bool
+        elif a.as_sym_bools is not None:
+            return a.as_sym_bools
+        elif a.as_graph is not None:
+            return None
+        elif a.as_optional_tensors is not None:
+            return a.as_optional_tensors
+        elif a.as_custom_obj is not None:
+            return None
+        else:
+            raise AssertionError(f"Unknown argument type: {a}")
+
+    # Stage 1: Reorder named items.
```

Comment:
I don't think it's generally safe to reorder nodes, because it affects the memory footprint of the program, and may affect numerics as well.

Reply:
> affects the memory footprint of the program

Hmm I thought we should still decouple frontend and backend at this layer.

Reply:
Ha! Yes, but due to how eager mode works I consider the ordering of operations as much of the semantic meaning of the graph as the operations themselves.